compiler/rustc_mir/src/const_eval/eval_queries.rs

```diff
@@ -30,6 +30,13 @@ fn eval_body_using_ecx<'mir, 'tcx>(
     body: &'mir mir::Body<'tcx>,
 ) -> InterpResult<'tcx, MPlaceTy<'tcx>> {
     debug!("eval_body_using_ecx: {:?}, {:?}", cid, ecx.param_env);
+    assert!(
+        cid.promoted.is_some()
+            || matches!(
+                ecx.tcx.hir().body_const_context(def_id),
+                Some(ConstContext::Const | ConstContext::Static(_))
```

Comment:
you'll need to add imports for `ConstContext`, it's in `rustc_hir` I believe.

You also need to use `cid.instance.def_id()` instead of just `def_id`, I didn't realize we don't have a variable for the `DefId` anymore.

Reply:
oh i completely forgot running ./x.py check before pushing again :D. thank you

Reply:
There's also an assertion below that you can drop (the "Assert all args" part).

Reply:
also `body_const_context` expects an `LocalDefId` instead of a `DefId`

Reply:
ah yes, but this can only ever be a local def id, so you can run `.expect_local()` on it. If It's not local, then we screwed up the query system :laughing:

Reply:
Oh this is a LocalId from `rustc_span`. I thought it was from `rustc_mir` and looking at the docs for that.

Reply:
> There's also an assertion below that you can drop (the "Assert all args" part).
i did not understand what you mean? Do you mean i can drop `cid.promoted.is_some()` part?

Reply:
No, further down. Ralf meant that you can remove the entire for loop at `// Assert all args (if any) are zero-sized types; `eval_body_using_ecx` doesn't`

Reply:
Since we now can't call this on functions at all, we won't ever have arguments, so there's no arguments to check. Thus the loop already does nothing

Reply:
> ah yes, but this can only ever be a local def id, so you can run .expect_local() on it. If It's not local, then we screwed up the query system

i run the test suite and got a panic saying def_id isn't local 😕